src/toil/jobStores/aws/jobStore.py

```diff
@@ -511,103 +525,102 @@ def read_shared_file_stream(self, shared_file_name, encoding=None, errors=None):
             if e.response.get('ResponseMetadata', {}).get('HTTPStatusCode') == 404:
                 raise NoSuchFileException(shared_file_name)
 
-    def delete_file(self, file_id):
-        """Only delete the reference."""
-        self.s3_client.delete_object(Bucket=self.bucket_name, Key=f'{self.metadata_key_prefix}{file_id}')
-        self.s3_client.delete_object(Bucket=self.bucket_name, Key=f'{self.content_key_prefix}{file_id}')
+    def delete_file(self, file_id: str) -> None:
+        full_s3_key = self.existing_full_s3_key_from_file_id(file_id)
+        self.s3_client.delete_object(Bucket=self.bucket_name, Key=full_s3_key)
```

Comment:
`delete_file` is required to be idempotent:
https://github.com/DataBiosphere/toil/blob/9f0d4580523d8f9f00a76feff252ab450a06c938/src/toil/jobStores/abstractJobStore.py#L1535-L1536
But `existing_full_s3_key_from_file_id` will raise an error if no such key exists, so you won't actually be able to safely call the delete twice.
Probably the error needs to be handled here.